### ISSUE TITLE
incompatible with numpy>=2

### ISSUE THREAD

**Issue by adrinjalali** [zillow/quantile-forest#58]
With numpy2:

```py
>>> from quantile_forest._quantile_forest_fast import QuantileForest
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/adrin/micromamba/envs/numpy2/lib/python3.12/site-packages/quantile_forest/__init__.py", line 26, in <module>
    from ._quantile_forest import ExtraTreesQuantileRegressor, RandomForestQuantileRegressor
  File "/home/adrin/micromamba/envs/numpy2/lib/python3.12/site-packages/quantile_forest/_quantile_forest.py", line 54, in <module>
    from ._quantile_forest_fast import QuantileForest, generate_unsampled_indices
  File "quantile_forest/_quantile_forest_fast.pyx", line 1, in init quantile_forest._quantile_forest_fast
ValueError: numpy.dtype size changed, may indicate binary incompatibility. Expected 96 from C header, got 88 from PyObject
```

**Comment by adrinjalali** [comment]
Encountered this while fixing `skops`'s CI

**Comment by reidjohnson** [comment]
@adrinjalali Thanks for reporting, will work on a fix!

**Comment by adrinjalali** [comment]
Thanks @reidjohnson . Please let me know when you do a release with numpy2.

**Comment by reidjohnson** [comment]
@adrinjalali This should now be fixed in the latest release, version `1.3.7`.
